tests/Microsoft.Identity.Test.Unit.net45/CacheTests/UnifiedSchemaValidationTests.cs

```diff
@@ -0,0 +1,878 @@
+                ["credential_type"] = "IdToken",
+                ["environment"] = "login.microsoftonline.com",
+                ["home_account_id"] = "9f4880d8-80ba-4c40-97bc-f7a23c703084.f645ad92-e38d-4d1a-b510-d1b09a74a8ca",
+                ["realm"] = "f645ad92-e38d-4d1a-b510-d1b09a74a8ca",
+                ["client_id"] = "b6c69a37-df96-4db0-9088-2ab96e1d8215",
+                ["client_info"] = "eyJ1aWQiOiI5ZjQ4ODBkOC04MGJhLTRjNDAtOTdiYy1mN2EyM2M3MDMwODQiLCJ1dGlkIjoiZjY0NWFkOTItZTM4ZC00ZDFhLWI1MTAtZDFiMDlhNzRhOGNhIn0"
+            };
+
+            AssertAreJsonStringsEquivalent(expectedJsonObject.ToString(), credential.ToJsonString());
+
+            MsalIdTokenCacheKey key = credential.GetKey();
+
+            string expectedServiceKey = "idtoken-b6c69a37-df96-4db0-9088-2ab96e1d8215-f645ad92-e38d-4d1a-b510-d1b09a74a8ca-";
+            Assert.AreEqual(expectedServiceKey, key.iOSService);
+
+            string expectedAccountKey = "9f4880d8-80ba-4c40-97bc-f7a23c703084.f645ad92-e38d-4d1a-b510-d1b09a74a8ca-login.microsoftonline.com";
+            Assert.AreEqual(expectedAccountKey, key.iOSAccount);
+
+            string expectedGenericKey = "idtoken-b6c69a37-df96-4db0-9088-2ab96e1d8215-f645ad92-e38d-4d1a-b510-d1b09a74a8ca";
+            Assert.AreEqual(expectedGenericKey, key.iOSGeneric);
+
+            Assert.AreEqual(2003, key.iOSType);
+        }
+
+        [TestMethod]
+        public void TestSchemaComplianceForRefreshToken_WhenMSSTSResponse_WithAADAccountAndFociClient()
+        {
+            // TODO: this test is failing because we don't create a refresh token AND a family refresh token in .NET.  So the keys are wrong since this 
+            // will become a FRT.  Need to align this with iOS folks.
+
+            var credential = new MsalRefreshTokenCacheItem(MsalEnvironment, ClientId, CreateAadTestTokenResponseWithFoci(), AadTenantId);
+
+            // TODO: ios tests do not have client_info key in them.  Should they?
+            var expectedJsonObject = new JObject
+            {
+                ["secret"] = "<removed_rt>",
+                ["credential_type"] = "RefreshToken",
+                ["environment"] = "login.microsoftonline.com",
+                ["home_account_id"] = "9f4880d8-80ba-4c40-97bc-f7a23c703084.f645ad92-e38d-4d1a-b510-d1b09a74a8ca",
+                ["client_id"] = "b6c69a37-df96-4db0-9088-2ab96e1d8215",
+                ["client_info"] = "eyJ1aWQiOiI5ZjQ4ODBkOC04MGJhLTRjNDAtOTdiYy1mN2EyM2M3MDMwODQiLCJ1dGlkIjoiZjY0NWFkOTItZTM4ZC00ZDFhLWI1MTAtZDFiMDlhNzRhOGNhIn0",
+                ["family_id"] = "1",
+            };
+
+            AssertAreJsonStringsEquivalent(expectedJsonObject.ToString(), credential.ToJsonString());
```

Comment:
I probably didn't fully implement the FRT serialization on iOS because FOCI is disabled on iOS. In hindsight, I should have had this throw a NotImplException. Should be simple to fix - see how we create the key for non-iOS.

Reply:
Do we actually generate a refresh token AND a FRT item in .NET?  The tests olga has have a RefreshToken and a FamilyRefreshToken as separate items.  This test is returning refreshtoken-1-- which is what i'd expect for a family token and not to get this at all.  i'm not sure why this is a test since we shouldn't have two items.

Reply:
Updated comments in code and I'm ignoring this test for now since we don't support the scenario in .NET of writing both.